pages/constructor-encoder.page.tsx

```diff
@@ -0,0 +1,171 @@
+  };
+
+  const setSingleError = (i: number, error: string) => {
+    setErrors(errors.map((oldError, id) => (i === id ? error : oldError)));
+  };
+
+  const validateInput = (i: number, value: string, type: string) => {
+    setSingleError(i, '');
+
+    if (type.startsWith('uint')) {
+      try {
+        BigNumber.from(value);
+      } catch (e) {
+        setSingleError(i, (e as any).reason);
+      }
+    } else if (type === 'address') {
+      if (!isValidAddress(value)) {
+        setSingleError(i, 'invalid address');
+      }
+    } else {
+      setSingleError(i, 'unsupported type');
+    }
+  };
+
+  return (
+    <ToolLayout>
+      <header className="mb-6 flex items-center gap-3 align-middle">
+        <DecoderSvg width={30} height={30} alt="deth calldata decoder icon" />
+        <h3 className="text-sm text-deth-gray-300 sm:text-xl">Encoders /</h3>
+        <h3 className="text-sm text-deth-pink sm:text-xl">
+          Constructor Encoder
+        </h3>
+      </header>
+
+      <label htmlFor="abi" className="pt-2 text-lg font-bold">
+        <p>ABI</p>
+      </label>
+
+      <textarea
+        id="abi"
+        aria-label="text area for constructor abi"
+        value={rawAbi || ''}
+        placeholder="e.g function transferFrom(address, ..)"
+        className="flex h-48 w-full break-words rounded-b-md border-t-0
+            border-deth-gray-600 bg-deth-gray-900 p-5"
+        onChange={(event: ChangeEvent<HTMLTextAreaElement>) => {
+          onAbiChange(event.target.value);
+        }}
+      />
+      <section className="mb-3">
+        {iface &&
+          iface.deploy.inputs.map((input, i) => (
+            <section className="flex items-center gap-2" key={input.name}>
+              <div className="flex flex-1 flex-col">
+                <label className="pb-2" htmlFor={`${input.name}`}>
+                  <div>
+                    <b className="text-purple-600">{input.type}</b>{' '}
+                    <b>{input.name}</b>
+                  </div>
+                </label>
+
+                <Input
+                  id={`${input.name}`}
+                  type="text"
+                  value={values[i]}
+                  error={errors[i]}
+                  placeholder="e.g 0x0..."
+                  className="mb-2 mr-auto h-10 w-3/5 rounded-md border border-deth-gray-600 bg-deth-gray-900 text-sm focus:outline-none"
+                  onChange={(event: ChangeEvent<HTMLInputElement>) => {
+                    const value = event.target.value;
+                    setSingleValue(i, value);
+                    validateInput(i, value, input.type);
+                  }}
+                />
+              </div>
+            </section>
+          ))}
+      </section>
+
+      <Button
+        onClick={onEncodeClick}
+        disabled={encodeButtonDisabled}
+        title={encodeButtonDisabled ? 'Please fill in the calldata' : undefined}
+      >
+        Decode
+      </Button>
+
+      <section className="pt-4">
```

Comment:
Let's stick to the rest of the layout of the other tools. There should be an empty darkfield, even if there is no encoded output yet.